rtsp.go

```diff
@@ -198,9 +203,48 @@ func (rc *rtspCamera) reconnectClient() (err error) {
 	return nil
 }
 
+// initMJPEG initializes the MJPEG decoder and sets up the client to receive JPEG frames.
+func (rc *rtspCamera) initMJPEG(tracks media.Medias, baseURL *url.URL) error {
+	var mjpegFormat *formats.MJPEG
+	track := tracks.FindFormat(&mjpegFormat)
+	if track == nil {
+		rc.logger.Warn("tracks available")
+		for _, x := range tracks {
+			rc.logger.Warnf("\t %v", x)
+		}
+		return errors.New("MJPEG track not found")
+	}
+
+	_, err := rc.client.Setup(track, baseURL, 0, 0)
+	if err != nil {
+		return err
+	}
+
+	mjpegDecoder, err := mjpegFormat.CreateDecoder2()
+	if err != nil {
+		return errors.Wrap(err, "error creating MJPEG decoder")
+	}
+
+	rc.client.OnPacketRTP(track, mjpegFormat, func(pkt *rtp.Packet) {
+		frame, _, err := mjpegDecoder.Decode(pkt)
+		if err != nil {
+			return
```

Comment:
I'm not sure why there are so many garbage frames that we just ignore the error here. I am mirroring https://github.com/viamrobotics/rdk/blob/0a7f6b23851f8c04f82ffa5583f600668e5e2e3a/components/camera/rtsp/rtsp.go#L171

I also checked and even if we ignore the error here, we are still getting a full 60fps stream on chrome webrtc internals